hadoop-ozone/ozonefs-shaded/pom.xml

```diff
@@ -120,38 +152,67 @@
                 <relocation>
                   <pattern>org</pattern>
                   <shadedPattern>${shaded.prefix}.org</shadedPattern>
-                  <includes>
-                    <include>org.yaml.**.*</include>
-                    <include>org.sqlite.**.*</include>
-                    <include>org.tukaani.**.*</include>
-                    <include>org.bouncycastle.**.*</include>
-                    <include>org.rocksdb.**.*</include>
-                    <include>org.apache.commons.cli.**.*</include>
-                    <include>org.apache.commons.compress.**.*</include>
-                    <include>org.apache.commons.codec.**.*</include>
-                    <include>org.apache.commons.beanutils.**.*</include>
-                    <include>org.apache.commons.collections.**.*</include>
-                    <include>org.apache.commons.digester.**.*</include>
-                    <include>org.apache.commons.io.**.*</include>
-                    <include>org.apache.commons.logging.**.*</include>
-                    <include>org.apache.commons.validator.**.*</include>
-                    <include>org.apache.commons.lang3.**.*</include>
-                    <include>org.sqlite.**.*</include>
-                    <include>org.apache.thrift.**.*</include>
-                  </includes>
+                  <excludes>
+                    <exclude>org.apache.hadoop.**</exclude>
+                    <exclude>org.apache.log4j.**</exclude>
+                    <exclude>org.apache.ozone.**</exclude>
+                    <exclude>org.apache.ratis.**</exclude>
+                    <!-- These packages are the ones that are not shaded in the hadoop release package.
+                      To prevent conflicts, we do not include these packages in the ozone client release package.
+                      Ozone will use hadoop / JVM to provide these dependencies. -->
+                    <exclude>org.bouncycastle.**</exclude>
+                    <exclude>org.ietf.jgss.*</exclude>
+                    <exclude>org.omg.**</exclude>
+                    <exclude>org.slf4j.**</exclude>
+                    <exclude>org.w3c.dom.**</exclude>
+                    <exclude>org.xerial.snappy.**</exclude>
+                    <exclude>org.xml.sax.**</exclude>
+                    <exclude>org.wildfly.**</exclude>
+                  </excludes>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.ratis</pattern>
```

Comment:
if we shade ratis on the `<pattern>org</pattern>` , will cause exception for the ratis file in the `META-INF`

![image](https://github.com/user-attachments/assets/f1f87737-ce52-4a9d-a7a1-75c1188d5d6f)

put ratis shade in a Independent unit can fix this issue

The normal ratis file
![image](https://github.com/user-attachments/assets/df02c3a4-9559-440c-bc3c-8bb8c0357e04)